schema.go

```diff
@@ -25,6 +25,10 @@ func (nullDefaultType) MarshalJSON() ([]byte, error) {
 var nullDefault nullDefaultType = struct{}{}
 
 var (
+	arrayReserved = []string{ // same as schemaReserved, but allows logicalType
+		"doc", "fields", "items", "name", "namespace", "size", "symbols",
+		"values", "type", "aliases", "precision", "scale",
+	}
```

Comment:
The only thing the spec says about extra attributes is this:

> Attributes not defined in this document are permitted as metadata, but must not affect the format of serialized data.

So preventing the use of these for _all_ types (even types for which these properties are not defined) seems a bit draconian. For example, it seems like arrays should be free to use properties named "fields", "precision", "scale", etc since they are not defined for the array type (and obviously wouldn't impact serialization).

Reply:
(BTW, I made this change in order for the tests to pass. I didn't catch them in original formulation because the code I am working on, in tandem with this pull request, was still providing the schemas as a JSON string, and the first iteration of this branch didn't round-trip that through `avro.Schema` but rather just called `json.Compact` on it).

Reply:
If this change is too controversial right now, I can back it out. I just realized I have a work-around for my current use case via a custom schema marshaler.

In the meantime, I'm noticing there are some other issues with how much this package discards from the original schema that will cause other issues with Iceberg. In particular, Iceberg also defines a logical type "timestamp-nanos" for longs and "uuid" for fixed[16] which would not survive being round-tripped into an `avro.Schema` and back to JSON right now. Looks like this issue has already been reported in #435.

Reply:
To be clear, this was a shortcut that so far no one noticed. I have no issue with adding a specific set for arrays. The intent was not to be draconian, but to stop these fields from being in properties and being parsed at the same time. 

Since this, the parsing has changed and there is really no need for this anymore, as props cannot encroach on reserved fields anymore, given how parsing now works, but I have not had the time to refactor this out.

Reply:
This is admittedly a hack that is only sufficient to get the test to pass, so something else is really warranted.

Here's what I'll do: For now, I'll undo this in this branch (and reformulate the test so it passes without this). But then I will follow-up with a separate pull request that addresses this more thoroughly and systematically (and would resolve #435).

Reply:
I am also looking at this: https://github.com/hamba/avro/pull/468